src/sentry/relay/config/metric_extraction.py

```diff
@@ -519,16 +519,17 @@ def _convert_aggregate_and_query_to_metrics(
     Extra metric specs will be returned if we need to maintain various versions of it.
     This makes it easier to maintain multiple spec versions when a mistake is made.
     """
-    try:
-        # We can avoid injection of the environment in the query, since it's supported by standard, thus it won't change
-        # the supported state of a query, since if it's standard, and we added environment it will still be standard
-        # and if it's on demand, it will always be on demand irrespectively of what we add.
-        if not should_use_on_demand_metrics(dataset, aggregate, query, groupbys, prefilling):
-            return None
 
-        metric_specs_and_hashes = []
-        # Create as many specs as we support
-        for spec_version in OnDemandMetricSpecVersioning.get_spec_versions():
+    # We can avoid injection of the environment in the query, since it's supported by standard, thus it won't change
+    # the supported state of a query, since if it's standard, and we added environment it will still be standard
+    # and if it's on demand, it will always be on demand irrespectively of what we add.
+    if not should_use_on_demand_metrics(dataset, aggregate, query, groupbys, prefilling):
+        return None
+
+    metric_specs_and_hashes = []
+    # Create as many specs as we support
+    for spec_version in OnDemandMetricSpecVersioning.get_spec_versions():
+        try:
```

Comment:
Moving the try block under the for loop since I do not want a failed metric spec from generating the remaining specs.